contents/testing/integration-testing.md

```diff
@@ -0,0 +1,214 @@
+
+## What is integration testing?
+
+Integration testing is a part of software testing where individual parts of your application are combined and tested as a group. This usually happens after the [<tooltip content="Testing of individual components within a system">unit testing stage</tooltip>](http://softwaretestingfundamentals.com/unit-testing/) and before the [<tooltip content="Evaluation of the software against requirements gathered from users and system specifications">system testing stage</tooltip>](https://www.guru99.com/system-testing.html) and it aims to discover faults related to the interaction between components.
+
+Imagine that you have built an online web ordering system and a sample architecture diagram of your application is shown as follows:
+
+<center>
+
+![Sample architecture diagram](integration-test/sample-architecture.png "Sample architecture diagram")
+</center>
+
+When focusing on the food ordering aspect of this system, a unit test on the `Order Summary` could look like this:
+```java
+@Test
+public void isOrderSummaryCorrect(OrderSummary orderSummary) {
+	assertTrue(orderSummary.contains("cheese burger"));
+}
+```
+
+After unit testing of individual modules are completed, an integration test on `Order Food` and `Order Summary` could look like this:
+
+```java
+public class FoodOrderingIntegrationTest {
+	public void testFoodOrderRecorded() {
+		FoodOrder foodOrder1 = new FoodOrder("cheese burger");
+		FoodSummary foodSummary = new FoodSummary();
+		foodSummary.add(foodOrder);
+		
+		isOrderSummaryCorrect(foodSummary);
+	}
+}
+```
+
+The system test is used to ensure that the entire application has all the necessary working functionalities such as ordering, paying and generating reports in different scenarios such as this:
+
+- A customer ordered 3 items, made payment using his E-Wallet and generated a report to save the order record.
+
+## Why is integration testing important?
+
+In a software application, each software module is usually designed and unit tested by different programmers.
+Since these programmers may work in isolation and have different understandings of the software requirements, integration testing is necessary to ensure that software modules work in unity and expose any faults in the interaction between different units.
+
+Suppose you are developing the `Order Food` component as mentioned previously and you allow each user to submit up to 10 orders each time, if the `Order Summary` component can only record a maximum of 5 orders each time, 5 orders by different users could be lost, resulting in problems such as **data loss**, **inaccurate results** and even **system errors**.
+Integration tests find these discrepancies before any serious damage is done.
+
+Although system testing also help to uncover faults between modules, integration testing is still highly encouraged as it is **easier to test**, **isolate features better** and **runs faster** than a system test.
```

Comment:
> it (integration testing) is easier to test?

Doesn't sound right

Similarly, `isolate features better` doesn't match. Maybe say something like this form: `it is ....., able to better isolate the scope of features`?